examples/websocket_client/Client.cpp

```diff
@@ -0,0 +1,184 @@
+    {
+    public:
+        explicit ConsoleFactory(infra::BoundedString url, services::Tracer& tracer);
+
+    public:
+        virtual infra::BoundedString Url() const override;
+        virtual uint16_t Port() const override;
+        virtual void ConnectionEstablished(infra::AutoResetFunction<void(infra::SharedPtr<services::ConnectionObserver> client)>&& createdClientObserver) override;
+        virtual void ConnectionFailed(ConnectFailReason reason) override;
+
+    private:
+        infra::BoundedString url;
+        services::Tracer& tracer;
+        infra::SharedOptional<ConsoleClientConnection> consoleClientConnection;
+    };
+
+    ConsoleFactory::ConsoleFactory(infra::BoundedString url, services::Tracer& tracer)
+        : url(url)
+        , tracer(tracer)
+    {}
+
+    infra::BoundedString ConsoleFactory::Url() const
+    {
+        return url;
+    }
+
+    uint16_t ConsoleFactory::Port() const
+    {
+        return 80;
+    }
+
+    void ConsoleFactory::ConnectionEstablished(infra::AutoResetFunction<void(infra::SharedPtr<services::ConnectionObserver> client)>&& createdClientObserver)
+    {
+        tracer.Trace() << "Connection established";
+        tracer.Trace();
+        createdClientObserver(consoleClientConnection.Emplace());
+    }
+
+    void ConsoleFactory::ConnectionFailed(ConnectFailReason reason)
+    {
+        tracer.Trace() << "Connection failed";
+        exit(1);
+    }
+}
+
+int main(int argc, const char* argv[], const char* env[])
+{
+    std::string toolname = argv[0];
+    args::ArgumentParser parser(toolname + " is a tool to demonstrate a websocket client.");
+    args::Group arguments(parser, "Arguments:");
+    args::Positional<std::string> url(arguments, "url", "URL of a websocket", args::Options::Required);
+    args::Group flags(parser, "Optional flags:");
+    args::HelpFlag help(flags, "help", "Display this help menu.", { 'h', "help" });
+
+    try
+    {
+        parser.Prog(toolname);
+        parser.ParseCLI(argc, argv);
+
+        static hal::TimerServiceGeneric timerService;
+        static hal::SynchronousRandomDataGeneratorGeneric randomDataGenerator;
+        static main_::TracerOnIoOutputInfrastructure tracer;
+        static main_::NetworkAdapter network;
+
+        static services::ConnectionFactoryWithNameResolverImpl::WithStorage<1> connectionFactory(network.ConnectionFactory(), network.NameResolver());
+        static services::HttpClientConnectorWithNameResolverImpl<> clientConnector{ connectionFactory };
+        static infra::Creator<services::Stoppable, services::HttpClientWebSocketInitiation, void(services::WebSocketClientObserverFactory & clientObserverFactory, services::HttpClientWebSocketInitiationResult & result, hal::SynchronousRandomDataGenerator & randomDataGenerator)> httpClientInitiationCreator{ [](infra::Optional<services::HttpClientWebSocketInitiation>& value, services::WebSocketClientObserverFactory& clientObserverFactory,
```

Comment:
Something weird happened here with the formatting of the function signature template argument.